airflow/models/dag.py

```diff
@@ -2880,11 +2884,14 @@ def dags_needing_dagruns(cls, session: Session):
         # TODO[HA]: Bake this query, it is run _A lot_
         # We limit so that _one_ scheduler doesn't try to do all the creation
         # of dag runs
+
         query = (
             session.query(cls)
             .filter(
+                cls.max_active_runs != None,  # noqa
```

Comment:
```suggestion
```

We don't need this condition now I think, right?

Reply:
Yes. Will remove it